apps/exam/views.py

```diff
@@ -59,7 +62,14 @@ def post(self, request, exam_id):
 
     current_attempt = ExamAttempt.objects.create(
         user=request.user, exam=exam, duration_minutes=exam_duration_minutes)
-    for question in exam.question_set.all():
+
+    if exam.question_set.all().count() < exam.question_count:
```

Comment:
Would .count() work without .all()?

Optionally: assign a related name for the question_set.